internal/controllers/cluster/cluster_controller_phases.go

```diff
@@ -143,132 +141,149 @@ func (r *Reconciler) reconcileExternal(ctx context.Context, cluster *clusterv1.C
 		)
 	}
 
-	return external.ReconcileOutput{Result: obj}, nil
+	return obj, nil
 }
 
 // reconcileInfrastructure reconciles the Spec.InfrastructureRef object on a Cluster.
-func (r *Reconciler) reconcileInfrastructure(ctx context.Context, cluster *clusterv1.Cluster) (ctrl.Result, error) {
+func (r *Reconciler) reconcileInfrastructure(ctx context.Context, s *scope) (ctrl.Result, error) {
```

Comment:
One difference to before is that a lot more things have to work before we declare CP/InfraCluster ready. I assume this is intentional and okay?